### DISCUSSION TITLE
Индикация

### DISCUSSION THREAD

**Discussion by jhnf** [vsulako/AFFBWheel#49] [Ideas]
Собираю руль на расширителях портов MCP23017, возникла идея сделать индикацию. 
Там есть 16 портов, если их использовать например так: 8 светодиодов уровень педали газа, 7 - положение руля (чтобы 4 нечётный показывала центр), 1 - индикация обратной связи. Соответственно это будет третий расширитель, отдельно от двух которые идут на кнопки.
Если у автора нет решения постараюсь сам разработать, но хотелось бы услышать оценку этой идеи.

**Comment by vsulako** [comment]
Индикация вообще никакая не планировалась.
Места в прошивке мало, а функция чисто эстетическая. Да и сделать ее настраиваемой сложно. 

Но вообще сделать конечно можно. 

как-то так:
1) завести переменную:
```
MCP23017_BBI2C mcp23017_dsp;
```
2) в setup настройка:
```
mcp23017_dsp.begin(адрес) 
mcp23017_dsp.writeRegister16(0, 0); //включим режим output на всех пинах
```
3) и далее где-нибудь в mainLoop: 

```
uint16_t dsp=0; 
/*
 тут надо приготовить значение dsp - какие пины включать
 wheel.analogAxes[0]->value - значение педали газа
 wheel.axisWheel->value - значение оси руля
 force - значение ОС
*/

mcp23017_dsp.writeRegister16(MCP23017_RGPIOA, dsp); //отправляем значение
```

**Reply by jhnf** [reply]
Спасибо, всё получилось. Памяти хватило :)
